src/test/java/api/support/matchers/JsonObjectMatcher.java

```diff
@@ -13,13 +12,45 @@
 import org.hamcrest.Matcher;
 import org.hamcrest.TypeSafeDiagnosingMatcher;
 
+import com.jayway.jsonpath.JsonPath;
+import com.jayway.jsonpath.PathNotFoundException;
+
 import io.vertx.core.json.JsonObject;
 
-public class JsonObjectMatcher extends TypeSafeDiagnosingMatcher<JsonObject> {
+public class JsonObjectMatcher {
 
   public static <T> Matcher<JsonObject> hasJsonPath(String jsonPath, Matcher<T> valueMatcher) {
-    return new JsonObjectMatcher(Collections.singletonList(
-      com.jayway.jsonpath.matchers.JsonPathMatchers.hasJsonPath(jsonPath, valueMatcher)));
+    return new TypeSafeDiagnosingMatcher<JsonObject>() {
+      @Override
+      protected boolean matchesSafely(JsonObject item, Description mismatchDescription) {
+        T actual;
+        try {
+          actual = JsonPath.parse(item.toString()).read(jsonPath);
+        } catch (PathNotFoundException notFoundException) {
+          actual = null;
```

Comment:
why does this set the actual value to null rather than immediately failing?

Is this matcher being used to evaluate paths that shouldn't be present?

Reply:
Yes, this is introduced to be able to assert a null value.

Reply:
> Yes, this is introduced to be able to assert a null value.

By a `null` value, are you referring to a property that is not present in the JSON?

Could the intent be expressed more clearer if there was a specific matcher for properties being not present?

Reply:
Yes, I can remove this logic. There should be a `hasNoJsonPath` matcher from javaway library.

Reply:
Removed.